### PR TITLE
fix/sekoia_action_add_ioc_to_ioc_collection

### PR DESCRIPTION
**Pull request by clement-burtscher-sekoia**
## Description

Relates to https://github.com/SekoiaLab/integration/issues/1873

### Added

- Add concise docstrings to all methods in `Add IOC to IOC Collection`
- Extend `Add IOC to IOC Collection` tests with edge cases (nested flattening, blank values, empty list, HTTP failure path) to reach 100% coverage on `sekoiaio/intelligence_center/add_ioc_to_ioc_collection.py`

### Changed

- Anonymize IOC/IP test samples with RFC 5737 and documentation IPv6 ranges to keep examples non-identifying while preserving input shapes
- Refactor `Add IOC to IOC Collection` tests to use `pytest.mark.parametrize` for repeated success and IP validation scenarios
- Use Pydantic v2 `TypeAdapter` to validate IPv4/IPv6 inputs in `add_IP_action` while preserving existing action behavior

### Fixed

- Fail explicitly in `Add IOC to IOC Collection` action when IP indicators are invalid (including CIDR notation like `/32`) instead of silently succeeding without creating indicators
- Add stricter IP validation to prevent green runs with empty `Results`/`Errors` when no valid IPv4/IPv6 indicator is actually submitted

### Removed

- Remove `indicators_path` from `Add IOC to IOC Collection` action JSON arguments because it was declared but not supported by the Python implementation

## Summary by Sourcery

Tighten IP indicator handling in the Add IOC to IOC Collection action and extend its test coverage.

Bug Fixes:
- Make the Add IOC to IOC Collection action fail explicitly when IP indicators are invalid, including unsupported CIDR notation, instead of succeeding without creating indicators.
- Ensure the IP-addition flow errors when no valid IPv4/IPv6 indicators are provided to avoid green runs with empty results.

Enhancements:
- Use Pydantic TypeAdapter-based validation for IPv4/IPv6 indicators in add_IP_action while preserving existing behavior.
- Add concise docstrings to the Add IOC to IOC Collection action methods and centralize IP examples in the add_IP_action docstring.
- Refactor Add IOC to IOC Collection tests to use parameterized scenarios and to cover edge cases such as nested flattening, blank values, empty lists, and HTTP failure paths.
- Update IOC/IP test samples to anonymized RFC 5737/documentation IPv6 ranges.

Documentation:
- Record the Add IOC to IOC Collection action behavior and validation changes in the changelog as release 2.75.3.

Tests:
- Expand Add IOC to IOC Collection tests to achieve full coverage of sekoiaio/intelligence_center/add_ioc_to_ioc_collection.py, including success paths, validation failures, and HTTP error handling.

### CODE REVIEW

File: Sekoia.io/sekoiaio/intelligence_center/add_ioc_to_ioc_collection.py [hunk 1]
```diff
@@ -33,27 +40,47 @@ def flatten_and_validate(self, indicators):
...
         if ipv4:
             self.perform_request(ipv4, ioc_collection_id, "ipv4-addr.value", valid_for)
         if ipv6:
             self.perform_request(ipv6, ioc_collection_id, "ipv6-addr.value", valid_for)
 
     def run(self, arguments: dict):
+        """Resolve input indicators and dispatch the matching IOC creation flow."""
         indicator_type_mapping = {
             "domain": "domain-name.value",
             "url": "url.value",
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Sekoia.io/sekoiaio/intelligence_center/add_ioc_to_ioc_collection.py:86]: Fixed with commit [Apply changes requested by Copilot review](https://github.com/SEKOIA-IO/automation-library/pull/2998/changes/8d2931900ff045f508d3fa1a6c1dd4a840e6bcf0)

File: Sekoia.io/tests/test_add_ioc_2_ioc_collection.py [hunk 2]
```diff
@@ -3,268 +3,212 @@
...
-    return {
-        "indicators": ["8.8.8.8"],
-        "ioc_collection_id": "ioc-collection--00000000-0000-0000-0000-000000000000",
-        "indicator_type": "IP",
-    }
-
-
-def test_add_ioc(arguments_success):
+IOC_COLLECTION_ID = "ioc-collection--00000000-0000-0000-0000-000000000000"
+INDICATORS_TEXT_ENDPOINT = "http://fake.url/api/v2/inthreat/ioc-collections/" + IOC_COLLECTION_ID + "/indicators/text"
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Sekoia.io/tests/test_add_ioc_2_ioc_collection.py]: Fixed with commit [Apply changes requested by Copilot review](https://github.com/SEKOIA-IO/automation-library/pull/2998/changes/8d2931900ff045f508d3fa1a6c1dd4a840e6bcf0)

File: Sekoia.io/action_add_ioc_to_ioc_collection.json [hunk 3]
```diff
@@ -3,23 +3,23 @@
...
       "indicators": {
-        "description": "List of indicators to add to an IOC collection",
+        "description": "List of indicators to add to an IOC collection. For IP address type, provide plain IPv4 or IPv6 values (CIDR is not supported).",
         "type": "array"
       },
       "indicators_path": {
-        "description": "Path of the indicators file to add to an IOC collection",
+        "description": "Path to an indicators file to add to an IOC collection.",
         "type": "string"
       },
```
Comment (clement-burtscher-sekoia) [inline Sekoia.io/action_add_ioc_to_ioc_collection.json]: @mchupeau-sk 

Action "Add IOC to IOC Collection":
- JSON file: [Sekoia.io/action_add_ioc_to_ioc_collection.json](https://github.com/SEKOIA-IO/automation-library/blob/fix/sekoia_action_add_ioc_to_ioc_collection/Sekoia.io/action_add_ioc_to_ioc_collection.json)
- related Python file: [Sekoia.io/sekoiaio/intelligence_center/add_ioc_to_ioc_collection.py](https://github.com/SEKOIA-IO/automation-library/blob/fix/sekoia_action_add_ioc_to_ioc_collection/Sekoia.io/sekoiaio/intelligence_center/add_ioc_to_ioc_collection.py)

I saw that the `indicators_path` parameter was declared in the action JSON file, but not explicitly handled in the related action Python file/class

What I understand is that the corresponding file contains a list of indicators (maybe one indicator by line if it is a text file), allowing the user to set more easily the action configuration on the playbook, especially when he deals with large indicator lists or reusable datasets

So I have several questions on this parameter:
1) Should the `indicators_path` parameter be supported in the related action Python file/class?
2) If yes, where is this file expected to be stored at runtime, and how should a playbook user make it accessible to the action?
3) What file format is expected (JSON or plain text), and what exact structure should it follow?

**Comment by mchupeau-sk** [comment]
@clement-burtscher-sekoia can you also update the manifest.json file with the new version

File: Sekoia.io/action_add_ioc_to_ioc_collection.json (hunk 3 above)
Comment (clement-burtscher-sekoia) [reply Sekoia.io/action_add_ioc_to_ioc_collection.json]: @mchupeau-sk Do you have any thoughts on this?

Comment (mchupeau-sk) [reply Sekoia.io/action_add_ioc_to_ioc_collection.json]: @clement-burtscher-sekoia it's seems to be dead code honestly, I don't see where it's used, for me we can remove it

Comment (clement-burtscher-sekoia) [reply Sekoia.io/action_add_ioc_to_ioc_collection.json]: @mchupeau-sk Fixed with commit [Remove indicators_path parameter](https://github.com/SEKOIA-IO/automation-library/pull/2998/changes/28bdb10af64e67e9cc58ccc81730e42b190de4ac)

**Comment by clement-burtscher-sekoia** [comment]
> @clement-burtscher-sekoia can you also update the manifest.json file with the new version

@mchupeau-sk Fixed with commit [Upgrade module version](https://github.com/SEKOIA-IO/automation-library/pull/2998/changes/e756dda37a29d6b48c4d8640c2cca3f75828997c)

**Review by mchupeau-sk — APPROVED** [review]
LGFM
